catalyst/runners/self_supervised.py

```diff
@@ -167,18 +171,36 @@ def _process_batch(self, batch):
         return batch
 
     def _process_input(self, batch: Mapping[str, Any], **kwargs):
-        embedding1, projection1 = self.model(batch[f"{self._augemention_prefix}_left"], **kwargs)
-        embedding2, projection2 = self.model(batch[f"{self._augemention_prefix}_right"], **kwargs)
-        origin_embeddings, projection_origin = self.model(batch[self._input_key], **kwargs)
-        batch = {
-            **batch,
-            f"{self._projection_prefix}_left": projection1,
-            f"{self._projection_prefix}_right": projection2,
-            f"{self._projection_prefix}_origin": projection_origin,
-            f"{self._embedding_prefix}_left": embedding1,
-            f"{self._embedding_prefix}_right": embedding2,
-            f"{self._embedding_prefix}_origin": origin_embeddings,
-        }
+
+        if isinstance(self.model, (collections.Mapping, nn.ModuleDict)):
+            encoders = [(encoder_name, self.model[encoder_name]) for encoder_name in self.model]
+        else:
+            encoders = [("", self.model)]
```

Comment:
let's move it to `on_stage_start`
so we could set up some internal flag like `is_kv_model` once per stage rather than every batch

Reply:
Like [that](https://github.com/catalyst-team/catalyst/pull/1305/commits/c9c9b989b26c2bdc981d4d229ea0f68452dd870a)?